### PR TITLE
Add tests to travis CI

### PR DESCRIPTION
**Pull request by kchitalia**
The build will fail once this PR is merged. To get the build to pass, we need to fix the failing tests. Not adding the tests for RestAPI as that freezes the build completely. Once we fix the tests, will add that back in.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,6 +20,11 @@ script:
...
-
-   #run all tests
+   # run all tests
    - xctool test -workspace SalesforceMobileSDK.xcworkspace -scheme SalesforceSDKCommon -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO
+   - xctool test -workspace SalesforceMobileSDK.xcworkspace -scheme SalesforceSecurity -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO
+   - xctool test -workspace SalesforceMobileSDK.xcworkspace -scheme SalesforceOAuthTest -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO
+   - xctool test -workspace SalesforceMobileSDK.xcworkspace -scheme SalesforceSDKCoreUnitTestApp -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO
+   - xctool test -workspace SalesforceMobileSDK.xcworkspace -scheme SalesforceNetworkiOS -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO
+   - xctool test -workspace SalesforceMobileSDK.xcworkspace -scheme SmartSync -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO
+   - xctool test -workspace SalesforceMobileSDK.xcworkspace -scheme SalesforceHybridSDK -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO
```
Comment (wmathurin) [inline .travis.yml]: Two seems to be missing OCMockLibTests and SmartSyncTests

Comment (kchitalia) [inline .travis.yml]: The SmartSync one is the second last line from bottom. Is that incorrect?

Comment (wmathurin) [inline .travis.yml]: I stand corrected the two missing ones are SalesforceRestAPITests and OCMockLibTests

Comment (kchitalia) [inline .travis.yml]: The RestAPITests freezes the build completely, so I have excluded that for now. I'll add the OCMockLibTests.

**Comment by wmathurin** [comment]
I was looking at what the build failure looks like - it's not easy to tell what tests are failing (and the output is truncated). Is there a way to improve that - maybe run a grep on the raw output?

**Comment by kchitalia** [comment]
Yes, definitely we can do some better storing of test results so that its easy to make out whats failing.

File: .travis.yml (hunk 1 above)
Comment (bhariharan) [inline .travis.yml]: We need to add ```HybridPluginTestApp```, ```SalesforceNetworkiOSTests``` and ```SalesforceRestAPIUnitTestApp``` to this list. I think ```HybridPluginTestApp``` is the one that runs the hybrid plugin tests.

Comment (kchitalia) [inline .travis.yml]: Will add HybridPluginTestApp and SalesforceNetworkiOSTests. The SalesforceRestAPIUnitTestApp freezes the build completely. I was thinking we could add that in after the tests are fixed?

Comment (bhariharan) [inline .travis.yml]: Okay.

**Comment by kchitalia** [comment]
Pretty printing of logs coming in a separate pull request

**Pull request merged**